### ISSUE TITLE
New screw up from KSP 1.11.0 Editor was revealed by the 2.4.6.20 release

### ISSUE THREAD

**Issue by Lisias** [TweakScale/TweakScale#283]
When scaling radially attached parts with resources, from KSP 1.11.x (tested from 1.11.2 only) only the part that received the scaling have the resources scaled accordingly.

The clones retain the last Resource set.

This is not related to KSP-Recall, as on KSP 1.10.2 the problem didn't manifested itself.

See [this](https://forum.kerbalspaceprogram.com/index.php?/topic/179030-ksp-130-tweakscale-under-lisias-management-24620-2023-0126/&do=findComment&comment=4224775) and [this](https://forum.kerbalspaceprogram.com/index.php?/topic/179030-ksp-130-tweakscale-under-lisias-management-24620-2023-0126/&do=findComment&comment=4224802) post on Forum for the gory details.

**Comment by Lisias** [comment]
While closing #282 , I detected that on KSP >= 1.11.x the Auto Scale started to behave somewhat like this one.

I'm reasonably sure that Editor, from 1.11.x, screwed `this.part.symmetryCounterparts` somehow.

**Comment by Lisias** [comment]
There's a possible explanation about the thing blowing up only after the refactoring: Monkey Patching.

On [AirplanePlus I detected ](https://github.com/net-lisias-ksp/AirplanePlus/issues/10)that KSP is able to Monkey Patch 3rd parties add'ons (bad, really bad move). Full history on [Forum](https://forum.kerbalspaceprogram.com/index.php?/topic/140262-14x-18x-airplane-plus-r264-fixed-issuesgithub-is-up-to-date-dec-21-2019/&do=findComment&comment=4194248).

So we know for sure that KSP does Monkey Patching on 3rd parties Add'Ons without warning neither a way to turn it off.

So I'm guessing (and this is a wild guess, I admit) that perhaps on KSP 1.11.x they Monkey Patched TweakScale. And then, when I refactored it to get rid of that crappy escuse of bytes I was calling code, the Monkey Patching failed and exposed the problem inherent to KSP 1.11x and newer.

Still wondering what to do about.

**Comment by Lisias** [comment]
Quoting myself from [Forum](https://forum.kerbalspaceprogram.com/index.php?/topic/140262-14x-18x-airplane-plus-r264-fixed-issuesgithub-is-up-to-date-dec-21-2019/&do=findComment&comment=4194248), just in case:

> Well… That was a weird one, and one that got me worried for a couple days (and, so, I screwed up the scheduling).
>
> Right now, A+ needs Firespitter on some of the parts. Yes, I know that there're replacements, but I would need to ask the user to install more than one add'ons to do the FS job - and yet, it will not be good enough (I know what I'm talking, replacing Firespitter on KAX wans't satisfactory). But most people forget to install dependencies anyway, and so I shove on every add'on of mine a WatchDog that barks on you everytime you fire up KSP missing dependencies my add'ons need, and while coding it into A+ I got this weird message on my KSP.log:
>
```
[ERR 12:27:47.935] Exception handling event onGameSceneSwitchRequested in class Startup:System.NullReferenceException: Object reference not set to an instance of an object
  at AirplanePlus.MonkeyPatching.Execute () [0x00000] in <filename unknown>:0
  at AirplanePlus.Startup.OnGameSceneSwitchRequested (FromToAction`2 data) [0x00000] in <filename unknown>:0
  at EventData`1[GameEvents+FromToAction`2[GameScenes,GameScenes]].Fire (FromToAction`2 data) [0x00000] in <filename unknown>:0

[EXC 12:27:47.941] NullReferenceException: Object reference not set to an instance of an object
        AirplanePlus.MonkeyPatching.Execute ()
        AirplanePlus.Startup.OnGameSceneSwitchRequested (FromToAction`2 data)
        EventData`1[GameEvents+FromToAction`2[GameScenes,GameScenes]].Fire (FromToAction`2 data)
        UnityEngine.DebugLogHandler:LogException(Exception, Object)
        KSPe.Util.Log.UnityLogDecorator:UnityEngine.ILogHandler.LogException(Exception, Object)
        ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
        UnityEngine.Debug:LogException(Exception)
        EventData`1:Fire(FromToAction`2)
        HighLogic:SetLoadSceneEventsAndFlags(GameScenes, Boolean)
        HighLogic:LoadScene(GameScenes)
        <SetupSystem>c__Iterator0:MoveNext()
        UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
[LOG 12:27:47.942] [HighLogic]: =========================== Scene Change : From LOADING to MAINMENU =====================
```
>
> The Exception **was not raised** from my code, and the testbed I was using was a pretty clean one, nothing but my supporting add'ons (KSPe, my fork of MM, and some utils). And I know all that code I forked or created, I inspected them (or wrote them from scratch), and there's no `MonkeyPatching` class/method/string anywhere on my repositories.
>
> So this stunt appears to be coming from KSP itself. I reproduced it down to KSP 1.3.1, by the way. Yep, this is lingering with us for a very long time.
>
> So I had something injecting code into AirplanePlus' WatchDog and causing it to bork. How I would release it this way? So I spent the weekend trying to locate it, understanding it and then trying a workaround for it (without breaking some Forum rules about modding, what's a bit of a challenge). And then the damned thing vanished!
>
> A bit after 31th Oct 00:00 AM (don't remember if 00:40 or 1:40 or something like that), the problem ceased to happen out of the blue. Right now I'm guessing (and guessing wildly) that this may be related to some Easter Egg code that was never finished and was forgotten on the codebase. But, again, this is a wild guess because I don't really have the slightest clue about why in the Name of the Krakens this krap fell on me on the last weekend.

**Comment by Lisias** [comment]
Oukey… Things are getting hairy…

1) You need to escale a "cloned" part in order to the thing misbehave. If you scale the "original", things works fine. It's a small detail that I forgot, and spent some time wondering why I wasn't reproducing the problem.

2) If you open the PAWs, **everything just works** no matter what part you scale!

![screenshot31](https://github.com/TweakScale/TweakScale/assets/64334/ceecb2dc-a1e6-4c32-b13c-bdd342f14192)

3) If you close the PAWs, then the misbehaviour happens again!

**Comment by Lisias** [comment]
Loading and saving the Craft "fixes" the problem, as TweakScale recalculates everything on Loading.

Launching the thing also works, at least with Stock parts.

**Comment by Lisias** [comment]
This is pretty annoying, but not a killer bug - saving and loading, or just launching the thing worked for me as a work around.

So I will postpone this one and priorize Companion's new features.

**Comment by Lisias** [comment]
Aaaaaannddd…. **THE FUCKING PROBLEM VANISHED OUT OF THE BLUE**.

Just it, the problem suddenly is no more. Don't know how, don't know why.

The only difference fro the last time I tried it is that I did it on my Acceptance Test Bed based on KSP 1.12.5 and with everything I **officially** maintain updated, in special https://github.com/net-lisias-ksp/KSPe , https://github.com/net-lisias-ksp/ModuleManager and https://github.com/net-lisias-ksp/ModuleManagerWatchDog and https://github.com/net-lisias-ksp/KSP-Recall .

Now… Why a completely unrelated changes (probably on Recall?) "fixed" this problem is, frankly, a complete mystery for me at this time (besides I have some suspicions about the reason).
